File: src/app/api/responses/item-response.ts
import {IItemData} from '../shared/item-data';
import {IActiveStatsModifier} from '../shared';
import {ItemTemplateResponse} from './item-template-response';

export interface ItemResponse {
    id: number;
    data: IItemData;
    modifiers?: IActiveStatsModifier[];
    containerId?: number;
    template: ItemTemplateResponse;
}

File: backend/src/Naheulbook.Web/Responses/ItemResponse.cs
using JetBrains.Annotations;
using Naheulbook.Shared.TransientModels;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Naheulbook.Web.Responses;

[PublicAPI]
public class ItemResponse
{
    public int Id { get; set; }
    public JObject Data { get; set; } = null!;
    public List<ActiveStatsModifier> Modifiers { get; set; } = null!;
    public int? ContainerId { get; set; }

    [JsonProperty("template")]
    public ItemTemplateResponse ItemTemplate { get; set; } = null!;
}